File: LibraryManagementSystem/src/main/java/com/dat/LibraryManagementSystem/service/PublisherService.java
package com.dat.LibraryManagementSystem.service;

import com.dat.LibraryManagementSystem.payload.dto.PublisherDTO;

import java.util.List;
import java.util.Map;

public interface PublisherService {
    PublisherDTO createPublisher(PublisherDTO dto) throws Exception;

    PublisherDTO updatePublisher(Long id, PublisherDTO dto);

    PublisherDTO getPublisherById(Long id);

    List<PublisherDTO> getAll();

    void deletePublisher(Long id);

    Map<String, Object> searchPublishers(String searchTerm, String address, int page, int size);
}

File: frontend/library-frontend/src/services/publisherService.js
import api from "./api";

export const publisherService = {
  searchPublishers: (params) => api.get("/publishers", { params }),
  createPublisher: (data) => api.post("/publishers", data),
  updatePublisher: (id, data) => api.put(`/publishers/${id}`, data),
  deletePublisher: (id) => api.delete(`/publishers/${id}`),
  getAllPublishers: () => api.get("/publishers"),
};
